### PR TITLE
Add memo and search attributes to child workflow

### PR DESCRIPTION
**Pull request by spmistry**

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/sync/SyncDecisionContext.java [hunk 1]
```diff
@@ -412,6 +412,8 @@ private ExecuteLocalActivityParameters constructExecuteLocalActivityParameters(
             .setWorkflowIdReusePolicy(options.getWorkflowIdReusePolicy())
             .setRetryParameters(retryParameters)
             .setCronSchedule(options.getCronSchedule())
+            .setMemo(options.getMemo())
```
Comment (longquanzheng) [inline src/main/java/com/uber/cadence/internal/sync/SyncDecisionContext.java:415]: look at line 387. I think we also need to do the same thing

**Comment by longquanzheng** [comment]
Also need a few line code to wire up the StartChildWorkflowExecutionParameters and StartChildWorkflowExecutionDecisionAttributes  https://github.com/uber/cadence-java-client/blob/master/src/main/java/com/uber/cadence/internal/replay/WorkflowDecisionContext.java#L156

File: src/main/java/com/uber/cadence/internal/sync/SyncDecisionContext.java (hunk 1 above)
Comment (spmistry) [reply src/main/java/com/uber/cadence/internal/sync/SyncDecisionContext.java:415]: updated

File: src/main/java/com/uber/cadence/internal/replay/WorkflowDecisionContext.java [hunk 2]
```diff
@@ -153,6 +155,20 @@ public void accept(Exception cause) {
...
+    Map<String, byte[]> memoMap = parameters.getMemo();
+    if (memoMap != null) {
+      memoMap.forEach((key, value) -> memo.putToFields(key, ByteBuffer.wrap(value)));
+      attributes.setMemo(memo);
+    }
+
+    SearchAttributes searchAttributes = new SearchAttributes();
+    Map<String, byte[]> searchAttributesMap = parameters.getSearchAttributes();
+    if (searchAttributesMap != null) {
+      searchAttributesMap.forEach((key, value) -> memo.putToFields(key, ByteBuffer.wrap(value)));
```
Comment (longquanzheng) [inline src/main/java/com/uber/cadence/internal/replay/WorkflowDecisionContext.java:168]: typo: `memo`->`searchAttributes`

Comment (spmistry) [reply src/main/java/com/uber/cadence/internal/replay/WorkflowDecisionContext.java]: fixed

**Review by longquanzheng — APPROVED** [review]

**Pull request merged**